### PR TITLE
process school and districts

### PR DESCRIPTION
**Pull request by malaffoon**
I still have to test against large volume so there might be some cleanup but ...

### CODE REVIEW

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/District.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package org.opentestsystem.rdw.ingest.processor.model;
```
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/District.java]: Moved District/School and their repos to RDW common.

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ImportPayloadSink.java [hunk 2]
```diff
@@ -4,17 +4,19 @@
 import org.springframework.messaging.SubscribableChannel;
 
 /**
- * MultiInputSink - Multiple Sink class for Spring Cloud processors that need more than one input channel
+ * ImportPayloadSink - multiple Sink class for import payload processor routing
  */
-public interface MultiInputSink {
-    String PACKAGE_INPUT = "packageInput";
-
+public interface ImportPayloadSink {
```
Comment (malaffoon) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ImportPayloadSink.java]: @rcordeau you okay with name change?

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultOrganizationProcessor.java [hunk 3]
```diff
@@ -0,0 +1,165 @@
+package org.opentestsystem.rdw.ingest.processor.service.impl;
```
Comment (malaffoon) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultOrganizationProcessor.java]: Guts of the change ...

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/District.java [hunk 4]
```diff
@@ -7,9 +7,8 @@
  * A district
  */
 public class District {
-    private final String name;
     private final String naturalId;
-    //TODO: this needs to be revisited for multi-state installation
```
Comment (agorina) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/District.java]: Did you remove the TODO for a reason? Are we now handling it somehow?

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultOrganizationProcessor.java [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+    public void process(final byte[] payload, final long importId) throws ImportException {
+        final OrgTreeParser parser = new OrgTreeParser();
+        parser.extractDistrictsAndSchools(payload);
+
+        for (final School school : parser.getSchools().values()) {
+            if (school.getDistrict() == null) {
+                logger.warn("skipping school id: {}, name: {} with unknown district", school.getNaturalId(), school.getName());
+                continue;
+            }
+            schoolRepository.upsert(school, importId);
```
Comment (agorina) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultOrganizationProcessor.java]: This repository caches the data. I think it is problematic for this use.

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ImportPayloadSink.java (hunk 2 above)
Comment (rcordeau) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/ImportPayloadSink.java]: Yeah, totally.

File: package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultOrganizationProcessor.java (hunk 5 above)
Comment (rcordeau) [inline package-processor/src/main/java/org/opentestsystem/rdw/ingest/processor/service/impl/DefaultOrganizationProcessor.java]: I do expect caching to be enabled when processing assessment packages and accommodation codes. I thought I had it enabled correctly but apparently not.

File: common/src/main/java/org/opentestsystem/rdw/ingest/common/model/District.java (hunk 4 above)
Comment (malaffoon) [inline common/src/main/java/org/opentestsystem/rdw/ingest/common/model/District.java]: No, it's just that there will be a lot of changes if/when we do a multi-state installation. Scattering TODO's through the code doesn't seem the best way to capture the work.

**Pull request merged**